tools/tsp-client/src/typespec.ts

```diff
@@ -114,38 +52,54 @@ export async function discoverMainFile(srcDir: string): Promise<string> {
   throw new Error(`No main.tsp or client.tsp found`);
 }
 
-export async function resolveImports(file: string): Promise<string[]> {
-  const imports: string[] = [];
-  const node = await parse(file);
-  for (const statement of node.statements) {
-    if (isImportStatement(statement)) {
-      imports.push(statement.path.value);
-    }
-  }
-  return imports;
-}
-
 export async function compileTsp({
   emitterPackage,
   outputPath,
   resolvedMainFilePath,
-  options,
+  additionalEmitterOptions,
+  saveInputs,
 }: {
   emitterPackage: string;
   outputPath: string;
   resolvedMainFilePath: string;
-  options: Record<string, Record<string, unknown>>;
+  additionalEmitterOptions?: string;
+  saveInputs?: boolean;
 }) {
-  Logger.debug(`Using emitter output dir: ${outputPath}`);
+  const parsedEntrypoint = getDirectoryPath(resolvedMainFilePath);
+  const { compile, NodeHost, getSourceLocation, resolveCompilerOptions } = await importTsp(parsedEntrypoint);
 
-  const parsedEntrypoint = path.parse(resolvedMainFilePath);
-  const { compile, NodeHost, getSourceLocation } = await importTsp(parsedEntrypoint.dir);
-  // compile the local copy of the root file
-  const program = await compile(NodeHost, resolvedMainFilePath, {
-    outputDir: outputPath,
+  const outputDir = resolvePath(outputPath);
+  const overrideOptions: Record<string, Record<string, string>> = {};
+  overrideOptions[emitterPackage] = {"emitter-output-dir": outputDir}
+  if (saveInputs) {
+    overrideOptions[emitterPackage]!["save-inputs"] = "true";
```

Comment:
I would save that object to a variable so you don't have to always `!`. Not sure you actually need to anyway

Reply:
It was strange to me as well that I got an error that overrideOptions[emitterPackage] might be undefined, because I am clearly setting an initial value for it on line 73